src/Microsoft.AspNet.Hosting/Program.cs

```diff
@@ -29,7 +29,9 @@ public void Main(string[] args)
             var tempConfig = tempBuilder.Build();
             var configFilePath = tempConfig[ConfigFileKey] ?? HostingJsonFile;
 
-            var appBasePath = _serviceProvider.GetRequiredService<IApplicationEnvironment>().ApplicationBasePath;
+            var appEnv = _serviceProvider.GetRequiredService<IApplicationEnvironment>();
+            appEnv.SetData("CommandLineArguments",args);
```

Comment:
so what you thoughts be if I created a IApplicationCommandLineConfiguration interface which is derived from IConfigurationRoot and used (IApplicationCommandLineConfiguration) new ConfigurationBuilder( ).AddCommandLine(args) to build an instance; which I then registered with service provider via the WebHostBuilder by using :--

            hostBuilder.UseServices(cnfgService =>
            {
                cnfgService.AddSingleton<IApplicationCommandLine>(_commandlineConfig);
            });